h/api/search.py

```diff
@@ -31,7 +34,41 @@ def _match_clause_for_uri(uristr):
     }
 
 
-def build_query(request_params):
+def scan(es_client, query, fields):
```

Comment:
If we didn't have to pass the client in I would understand the encapsulation argument. We would actually be hiding the details of the implementation.

But when our arguments include elasticsearch client instances and elasticsearch-formatted query objects I think the encapsulation is bogus anyway and it's better not to pretend.